.editorconfig

```diff
@@ -1,20 +1,371 @@
-# top-most EditorConfig file
+###############################################################################
+# EditorConfig is awesome: http://EditorConfig.org
+###############################################################################
+
+###############################################################################
+# Top-most EditorConfig file
+###############################################################################
 root = true
 
-[*.cs]
+###############################################################################
+# Set default behavior to:
+#   a UTF-8 encoding,
+#   Unix-style line endings,
+#   a newline ending the file,
+#   4 space indentation, and
+#   trimming of trailing whitespace
+###############################################################################
+[*]
+charset = utf-8
+end_of_line = lf
```

Comment:
The `end_of_line` default was changed to `lf` since that is what git will check it in as (all platforms/by default) and is what git will check it out as on Unix. It makes for better cross-platform consistency.

Reply:
Won't this make a mess of files if a user has set autocrlf=true in their git global settings?
Might want to add a gitattributes file to make sure they're checked out as lf https://www.git-scm.com/docs/gitattributes

Reply:
Good point. I hadn't looked at the `.gitattributes` file as part of this PR, will update.

Reply:
`.gitattributes` file was updated to explicitly call out `eol=lf` as the default and to explicitly specify various attributes for existing file types to ensure that it is handled correctly.

Reply:
I think you need to set bat/cmd to always be crlf or they won't run. Not at computer so can't test it, going on memory

Reply:
bat/cmd definitely work with lf line endings. I have it configured that way in all of my projects and did just test locally (and I don't believe it is Windows 10 specific).

Reply:
Perfect 😀